src/types/Patterns.ts

```diff
@@ -18,9 +18,22 @@
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
 
+import { getRequireHttpsUrlPattern } from 'services/EnvironmentService'
+
+// check the REQUIRE_HTTPS_URL_PATTERN environment variable, defaulting to 'true' if not set
+const requireHttpsUrlPattern = getRequireHttpsUrlPattern() === 'true'
+
 const DOMAIN =
   /([a-z0-9]|[a-z0-9][a-z0-9-]{0,61}[a-z0-9])(\.([a-z0-9]|[a-z0-9][a-z0-9-]{0,61}[a-z0-9])){1,10}/i
 const URLPATH = /(\/[a-z0-9-._~:/?#[\]@!$&'()*+,;=%]{0,500}){0,20}/
+// construct regex patterns for URL based on the REQUIRE_HTTPS_URL_PATTERN environment variable
+const urlPattern = new RegExp(
+  `^(http${requireHttpsUrlPattern ? 's' : 's?'})://(${
+    DOMAIN.source
+  })(:\\d{1,5})?(${URLPATH.source})?$`,
+  'i'
+)
+const prefixUrlPattern = requireHttpsUrlPattern ? /^https:/i : /^https?:/i
```

Comment:
sorry to bother again. I only noticed now that we are using this two times so I'd change lines 30-36 to (formatting ignored here):
```
const urlProtocol = requireHttpsUrlPattern ? 'https' : 'https?'
const prefixUrlPattern = new RegExp(`^${urlProtocol}:`, 'i')
const urlPattern = new RegExp(`^(${urlProtocol})://etc...$`, 'i')
```

Reply:
done